### ISSUE TITLE
Jetpack Photon: Add responsive images support

### ISSUE THREAD

**Issue by mjangda** [Automattic/vip-go-mu-plugins#587]
As of 4.4, WordPress can automatically generate and add the necessary markup for `srcset` support (see [wp_make_content_images_responsive](https://developer.wordpress.org/reference/functions/wp_make_content_images_responsive/)). However, by default that relies on image sizes being defined in `_wp_attachment_metadata`, which don't exist on Go (since all images are dynamically generated).

We can use a combination of the `wp_calculate_image_srcset_meta` filter + Jetpack Photon to fake our way through and add the the necessary pieces for this to work.

Rough idea of how to make this work:

Hook in to `wp_calculate_image_srcset_meta` and add the `sizes` we want included in our `srcset`. The resulting `metadata` array + `sizes` should look something like:

```
array (
    'width' => 1500,
    'height' => 977,
    'file' => '2017/01/image.jpg',
    'sizes' =>
    array (
      'thumbnail' =>
      array (
        'file' => 'image-150x98.jpg',
        'width' => 150,
        'height' => 98,
        'mime-type' => 'image/jpeg',
      ),
      'medium' =>
      array (
        'file' => 'image-300x195.jpg',
        'width' => 300,
        'height' => 195,
        'mime-type' => 'image/jpeg',
      ),
   ),
)
```

Note that the `file` for each size doesn't have to physically exist but for Jetpack's Photon filters we need to include it in the array and follow the standard naming convention. We can either reuse or copy [`WP_Image_Editor::generate_filename`](https://developer.wordpress.org/reference/classes/wp_image_editor/generate_filename/) for this.

We also may want a custom filter for VIPs to adjust the sizes included in the srcset.

**Comment by pkevan** [comment]
@mjangda I'm working on a patch for this. Any suggestion as to where the code should go? One of the `001-x` files or a new one `001-image-meta.php`?

**Comment by mjangda** [comment]
Probably best to keep as part of `a8c-files.php` for now?

**Comment by ivankristianto** [comment]
I see there is a fix for this issue at PR #648 
Do we have an eta when this could be release?

**Comment by mjangda** [comment]
See #860 and related Lobby post.
